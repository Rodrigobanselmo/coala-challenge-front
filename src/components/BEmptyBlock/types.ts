import { ReactNode } from "react";

export interface BEmptyBlockProps {
  text: string;
  endComponent?: ReactNode;
}
