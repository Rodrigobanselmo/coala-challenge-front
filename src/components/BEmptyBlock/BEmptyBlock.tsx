import React, { FC, Fragment } from "react";

import { BBookIcon } from "@/assets/icons/BBookIcon";
import {
  Box,
  Button,
  LinearProgress,
  Paper,
  TextField,
  Typography,
} from "@mui/material";

import { BEmptyBlockProps } from "./types";

export const BEmptyBlock: FC<BEmptyBlockProps> = ({ text, endComponent }) => {
  return (
    <Paper
      elevation={3}
      sx={{
        padding: 20,
        width: "100%",
        display: "flex",
        justifyContent: "space-between",
      }}
    >
      <Typography variant="body1" color="textSecondary">
        {text}
      </Typography>
      {endComponent}
    </Paper>
  );
};
