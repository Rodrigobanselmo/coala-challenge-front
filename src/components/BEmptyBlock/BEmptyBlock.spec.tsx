import React from "react";
import { render } from "@testing-library/react";
import { BEmptyBlock } from "./BEmptyBlock";
import { Button } from "@mui/material";

describe("BEmptyBlock", () => {
  it("matches the snapshot", () => {
    const { container } = render(
      <BEmptyBlock
        text="No books found"
        endComponent={<Button>Retry</Button>}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
