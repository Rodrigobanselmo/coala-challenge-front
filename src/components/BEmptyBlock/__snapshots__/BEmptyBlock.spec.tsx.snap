// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BEmptyBlock matches the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1bbcyej-MuiPaper-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
    >
      No books found
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Retry
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
